packages/fast-css-editor-react/app/app.tsx

```diff
@@ -0,0 +1,145 @@
+            }
+        },
+        editorExample: {
+            position: {
+                position: PositionValue.absolute,
+                left: 4,
+                top: 12
+            }
+        },
+        positionDetail: {
+            position: PositionValue.static
+        },
+        positionExample: {
+            position: PositionValue.absolute,
+            left: 1,
+            top: 5
+        }
+    };
+
+    public render(): JSX.Element {
+        return (
+            <Site formChildOptions={[]}>
+                <SiteTitle slot={"title"}>
+                    <SiteTitleBrand>FAST</SiteTitleBrand> CSS React editor
+                </SiteTitle>
+                <SiteCategory slot={"category"} name={"Components"}>
+                    <SiteCategory
+                        slot={"category"}
+                        name={"CSS Editor"}
+                        schema={CSSEditorSchema}
+                        component={CSSEditor}
+                    >
+                        <SiteCategoryDocumentation slot={"canvas-detail-view-documentation"}>
+                            <div>
+                                CSS Editor
+                            </div>
+                        </SiteCategoryDocumentation>
+                        <SiteCategoryItem
+                            slot={"canvas-example-view"}
+                            data={this.state.editorExample}
+                        />
+                        <SiteCategoryItem
+                            slot={"canvas-detail-view-example"}
+                            data={this.state.editorDetail}
+                        />
+                    </SiteCategory>
+                    <SiteCategory
+                        slot={"category"}
+                        name={"CSS Position"}
+                        schema={CSSPositionSchema}
+                        component={CSSPosition}
+                    >
+                        <SiteCategoryDocumentation slot={"canvas-detail-view-documentation"}>
+                            <div>
+                                CSS Position
+                            </div>
+                        </SiteCategoryDocumentation>
+                        <SiteCategoryItem
+                            slot={"canvas-example-view"}
+                            data={this.state.positionExample}
+                        />
+                        <SiteCategoryItem
+                            slot={"canvas-detail-view-example"}
+                            data={this.state.positionDetail}
+                        />
+                    </SiteCategory>
+                </SiteCategory>
+            </Site>
+        );
+    }
+
+    public componentWillMount(): void {
+        this.setState({
+            editorDetail: Object.assign({}, this.state.editorDetail, {onChange: this.handleEditorDetailUpdate}),
+            editorExample: Object.assign({}, this.state.editorExample, {onChange: this.handleEditorExampleUpdate}),
```

Comment:
Any reason you're not doing this in the constructor? Also, state isn't really readonly though you've assigned it as such. I think you should do all of the initial state setup in the constructor.